Client/mods/deathmatch/logic/luadefs/CLuaBlipDefs.cpp

```
@@ -62,353 +69,120 @@ void CLuaBlipDefs::AddClass(lua_State* luaVM)
-                CClientRadarMarker* pMarker =
-                    CStaticFunctionDefinitions::CreateBlip(*pResource, vecPosition, ucIcon, ucSize, color, sOrdering, usVisibleDistance);
-                if (pMarker)
-                {
-                    CElementGroup* pGroup = pResource->GetElementGroup();
-                    if (pGroup)
-                    {
-                        pGroup->Add(pMarker);
-                    }
-
-                    lua_pushelement(luaVM, pMarker);
-                    return 1;
-                }
-            }
-        }
+        m_pScriptDebugging->LogWarning(luaVM, SString("Blip size beyond %i is no longer supported (got %i). It will be clamped between 0 and %i.", MAX_BLIP_SIZE, size.value(), MAX_BLIP_SIZE));
+        size = MAX_BLIP_SIZE;
     }
-    else
-        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
-
-    lua_pushboolean(luaVM, false);
-    return 1;
-}
 
-int CLuaBlipDefs::CreateBlipAttachedTo(lua_State* luaVM)
-{
-    CClientEntity* pEntity = NULL;
-    // Default colors and size
-    unsigned char    ucIcon = 0;
-    int              iSize = 2;
-    SColorRGBA       color(255, 0, 0, 255);
-    int              iOrdering = 0;
-    int              iVisibleDistance = 16383;
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pEntity);
-    argStream.ReadNumber(ucIcon, 0);
-    argStream.ReadNumber(iSize, 2);
-    argStream.ReadNumber(color.R, 255);
-    argStream.ReadNumber(color.G, 0);
-    argStream.ReadNumber(color.B, 0);
-    argStream.ReadNumber(color.A, 255);
-    argStream.ReadNumber(iOrdering, 0);
-    argStream.ReadNumber(iVisibleDistance, 16383);
-
-    if (!CClientRadarMarkerManager::IsValidIcon(ucIcon))
+    CResource* resource = &lua_getownerresource(luaVM);
+    if (!resource)
     {
-        argStream.SetCustomError("Invalid icon");
+        m_pScriptDebugging->LogError(luaVM, "Couldn't find the resource element");
+        return false;
```

Small point, but, you can throw LuaFunctionError instead of logging it manually

---

I don't know why, but it shows warning in debugscript, not error. Maybe only in debug build?

---

If you dont provide additional parameter then it will throw warning by default.
To throw an error, pass `false` to the class